docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,345 @@
+---
+title: Board Pin Control
+sidebar_label: Pin Control
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+:::info
+This page exists to provide a guide to [Pin Control](https://docs.zephyrproject.org/3.5.0/hardware/pinctrl/index.html#pin-control) for ZMK users and designers. Refer to [Zephyr's page on Pin Control](https://docs.zephyrproject.org/3.5.0/hardware/pinctrl/index.html#pin-control) for elaboration and more details on any of the points raised here.
+:::
+
+A basic keyboard design as introduced in the [new shield guide](./new-shield.mdx) only uses its pins for the keyboard matrix. Many keyboard designs make use of advanced components or functionality, such as displays or shift registers. This results in the keyboard making use of communication protocols such as (but not limited to) SPI, I2C, or UART. Configuring pins for the usage of advanced functionality such as drivers for the previously named protocols is referred to as "Pin Control".
+
+:::warning
+The details of pin control can vary from vendor to vendor. An attempt was made to be as general as possible, but it isn't possible to cover all possible cases. The approaches for the nRF52840 and rp2040 MCUs/SoCs are documented in their entirety below. For other MCUs/SoCs, please refer to the [Zephyr documentation](https://docs.zephyrproject.org/3.5.0/index.html) and the examples and other files found in-tree of [ZMK](https://github.com/zmkfirmware/zmk/tree/main/app/boards) and [ZMK's fork of Zephyr](https://github.com/zmkfirmware/zephyr).
+:::
+
+## Boards, Shields, and Modules
+
+:::tip
+Some boards, such as the Seeed Studio Xiao, already have some of their pins configured for particular communication protocols. If this is the case, then you can use the pre-configured bus rather than configuring one manually (you merely need to identify the correct label for the bus node).
+:::
+
+Pin control is always defined for a _board_, never for a shield:
+
+- If you are defining a keyboard that is a _board_, then you will be editing and adding on to the files in your keyboard's folder as usual.
+- If your keyboard consists of a _board and a shield_, then you should define a new folder inside of your shield's folder called `boards`. You will need to add a `<board>.overlay` for each board to be used with said shield. Take for example the file structure of the "sofle" shield:
+  ```plaintext
+  boards/shields/sofle/
+  ├── boards/
+  │   ├── nice_nano.overlay
+  │   ├── nice_nano_v2.overlay
+  │   ├── nrfmicro_11.overlay
+  │   └── nrfmicro_13.overlay
+  └── <sofle shield defining files>
+  ```
+  Note that you will need to define a separate overlay _for each_ of the boards to be used with the shield.
+
+### External Shields
```

Comment:
How about we pose this section in a different way, to side step the issue: We could call this section "adding pinctrl to existing keyboards". This way, we can come at this from a user perspective where things don't have to be 100% compliant, where we can then e.g. recommend putting it in `config/shield.overlay` (I think that should work).

In such a case I also don't know if we'd recommend a PR; they could simply be bodging a display or something to their keyboards.

Reply:
Hmmm. I think we end up wanting both, though from past experience I think we're better off putting it in `config/shield.keymap`. If people want to configure the standard bus then they should PR, otherwise they should `.keymap`.

Reply:
The keymap is better because it overrides everything, but it isn't per side (and it seems bad to recommend the per-part keymap trick). The overlay is sometimes overridden by upstream but if the node wasn't defined at all it shouldn't be an issue?

Reply:
It turns out that my idea doesn't work after all, so I removed the section and added the forking note.